### PR TITLE
docs(legal): Updated agreements and policies for version 2

### PR DESCRIPTION
**Pull request by mffap**
With the introduction of our [improved cloud offering](https://zitadel.com/v2) we updated the terms of service, policies, and list of sub-processors. Amendments to the agreements and policies effective 15.7.2022 for all services. 

Most of these changes affect users who are using our improved cloud offering (aka. ZITADEL v2), but this update also contains a minor change to our support service terms. Users who continue to use our old offering (aka. ZITADEL v1) are bound under the hitherto existing terms and policies, which can be still found [here](https://docs-v1.zitadel.com/). As previously announced the service will be stopped by 31.12.2022, you can find additional information and information about migration [here](https://zitadel.com/v2).

## Change in sub-processors

On the commercial side the new offering introduces a new subscription model and options to enhance the base services, resulting in changes to the service description. On the technical side we introduced changes to our infrastructure that required us to change our subprocessors as follows: 
- Removed Cloudflare: We replaced Cloudflare with Google Services to remove the overall number of subprocessors and avoid TLS termination by an intermediary party for critical traffic to our edge network
- Removed Namecheap: We consolidated domain registration on Google
- Added [CockroachLabs](https://www.cockroachlabs.com/privacy/): We are using a dedicated CockroachDB Cloud instances for as our globally-distributed database. 

Depending on the use of our services, following sub-processors might apply: 
- Twillio: Must be enabled by configuration
- [Discord](https://discord.com/privacy): For using our public [chat](https://zitadel.com/chat)
- [Agolia ](https://www.algolia.com/policies/privacy/)search: For searching on our docs page

We ensured to follow the [guidelines on transborder data flows](https://www.edoeb.admin.ch/edoeb/en/home/data-protection/handel-und-wirtschaft/transborder-data-flows.html) for checking the admissibility of data transfers with reference to foreign countries.

## Summary of changes

See below for a summary of changes to the documents.

### Terms of Service

- Financial / Credit and payment: remove "in advance"
- Spelling
- Updated contact information
- Remove dedicated instance annex

### Service Descriptions

- Add service description for Cloud Services: We define terms used for billing and data location
- Add SAML to core services
- Update support service descriptions based on new offering
- Remove dedicated instance annex: part of the purchase order
- Remove annexes to the dedicated instance terms

### Terms and Conditions for Support Services

- This change adds a section about handling disputes about change of service in good faith. The change gives explicit right to customers to challenge modification.

#### Types of personal data
- Basic data: remove language; language data is not saved for every user, see profile information
- Profile data: remove unused birthday and address(es)
- Profile data: add nickname, display name
- Payment data: payment information

#### Types of personal data
- Basic data: remove language; language data is not saved for every user, see profile information
- Profile data: remove unused birthday and address(es)
- Profile data: add nickname, display name
- Payment data: payment information

#### Cookies
- Remove Cloudflare cookies: Cloud flare is no longer used for our services
- Add Customer portal cookies: These cookies are used for authenticating users visiting our customer portal

### CODE REVIEW

File: docs/docs/guides/customization/texts.md [hunk 1]
```diff
@@ -39,4 +39,4 @@ ZITADEL is available in the following languages
 
 A language is displayed based on your agent's language header. The default language is English.
 
-If you need support for a specific language we highly encourage you to [contribute translation files](/CONTRIBUTING.md) for the missing language.
+If you need support for a specific language we highly encourage you to [contribute translation files](../CONTRIBUTING.md) for the missing language.
```
Comment (fforootd) [inline docs/docs/guides/customization/texts.md:42]: I think this needs to be an external link like https://github.com/zitadel/zitadel/blob/v2-alpha/CONTRIBUTING.md

**Comment by fforootd** [comment]
There are some conflict to resolve btw.

File: docs/docs/guides/customization/texts.md (hunk 1 above)
Comment (mffap) [reply docs/docs/guides/customization/texts.md:42]: Out of context. This was due to a warning. Will try to remove it.

**Comment by mffap** [comment]
> There are some conflict to resolve btw.

Done

**Comment by mffap** [comment]
This PR was replaced by https://github.com/zitadel/zitadel/pull/3823

**Pull request closed without merging**